README.md

```diff
@@ -253,6 +253,27 @@ end;
 Now, if you run both programs, you'll see two `Hello World` messages, as the
 server sends the same message back to the client.
 
+## Using Mux with Revise.jl for Development
+
+Because Revise.jl cannot easily update variables, but functions, there is an easy alternative style to use Mux which works well with Revise.jl.
+
+The initial example is rewritten as follows. Removing the `@app` macro, which would define a variable, and using a function and the `Mux.App` wrapper instead.
+```julia
+using Mux
+
+test(req) = req |> mux(
+  Mux.defaults,
+  page(respond("<h1>Hello World!</h1>")),
+  page("/about",
+       probability(0.1, respond("<h1>Boo!</h1>")),
+       respond("<h1>About Me</h1>")),
+  page("/user/:user", req -> "<h1>Hello, $(req[:params][:user])!</h1>"),
+  Mux.notfound(),
+)
+serve(Mux.App(test))
+```
+Importantly, Revise.jl will only pick up file changes if a new line is executed in the REPL. So make sure to run the server in a background task like `serve` is doing by default.
```

Comment:
Thank you! How about this edit? I haven't tested the thing about pressing enter, I'm not sure I needed to do that in the past.

````suggestion
Revise.jl does not work well with Mux's `@app` macro because it tracks function definitions, not variables.

If you would like to use Revise.jl then you can do so by defining your web_app as a function and using the `Mux.App` wrapper, as we have done in the example below:

```julia
# my_web_app.jl
using Mux

my_web_app(req) = req |> mux(
  Mux.defaults,
  page(respond("<h1>Hello World!</h1>")),
  page("/about",
       probability(0.1, respond("<h1>Boo!</h1>")),
       respond("<h1>About Me</h1>")),
  page("/user/:user", req -> "<h1>Hello, $(req[:params][:user])!</h1>"),
  Mux.notfound(),
)

serve(Mux.App(my_web_app))
```

```julia-repl
julia> using Revise
julia> Revise.includet("my_web_app.jl")
julia> # Your web app is now running
julia> # Make some changes to my_web_app.jl and then press enter at the REPL to make Revise detect the changes
```
````

Reply:
The change sounds good. The part with the REPL is however very important... Revise won't revise if there is no change in the REPL.

I opened an [issue at Revise.jl](https://github.com/timholy/Revise.jl/issues/816) for this and the workaround is to use 
- either the default REPL trigger (if revise fails it will also show the output in the REPL), 
- or put a manual trigger inside the http serve loop (like linked in the issue)

Reply:
I just realized that you are still including the comment to press ENTER. Looks great!